### PR TITLE
Rename RoadNetwork to RoadContainer.

### PR DESCRIPTION
**Pull request by TheDuckCow**
Large change that makes the following updates, in preparation for intersections and new UI flow:

- Restructures the addon such that the top-level is cleaner, and distinct nodes that users add are all in a "nodes" folder (one outliar being the RoadSegment which we intentionally don't expose, but I still put it there since it is technically a node still and during runtime, game devs may want to interact with it).
- New icons! See below
- Placeholder classes for intersections and road manager, with no current functionality other than warning signs.
- Renaming of all references from RoadNetwork to RoadContainer.

Implements most of the following, but does not address how two Roads are connected together just yet.
https://github.com/TheDuckCow/godot-road-generator/issues/122

All tests pass: 12 passed 0 failed.  Tests finished in 1.3s.  1 orphan.

New icons! Could be further tweaked, but aligned more to the "red-orange = implied 3D node", plus using a shape that stylistically fits with the logo made by Kenney.

Screen Shot 2023-09-24 at 10 22 16 AM

Though worth also throwing up here "option B" which is something I had messed around with for a few minutes

Screen Shot 2023-09-23 at 10 07 53 AM

### CODE REVIEW

File: addons/road-generator/nodes/road_container.gd [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
 export(float) var density:float = 1.0  setget _set_density # Mesh density of generated segments.
-export(bool) var use_lowpoly_preview:bool = false  # Whether to reduce geo mid transform.
 
+# Generate procedural road geometry
+# If off, it indicates the developer will load in their own custom mesh + collision.
+export(bool) var create_geo := true setget _set_create_geo
+# If create_geo is true, then whether to reduce geo mid transform.
+export(bool) var use_lowpoly_preview:bool = false
+
+## Auto generated exposed variables dused to conneect this RoadContainer to
```
Comment (bdog2112) [inline addons/road-generator/nodes/road_container.gd:25]: Update words "dused to connect".

File: addons/road-generator/nodes/road_container.gd [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+## another RoadContainer.
+# connection_nodepaths = list of Nodepaths to pther RoadContainers, to indicate
```
Comment (bdog2112) [inline addons/road-generator/nodes/road_container.gd:27]: Update word "pther".

**Review by bdog2112 — COMMENTED** [review]
TESTED DEMO SCENE:
- Opened demo scene, allowed it to update, and saved it.
- Closed and re-opened demo scene. No errors upon reload.
- Selected and moved nodes. No errors.
- Noted that "RoadNetwork" node is now a "RoadContainer" and RoadPoints are now children of the RoadContainer.
- Closed demo scene.

TESTED NEW SCENE:
- Created new scene with "RoadContainer1" as root node. RoadContainer1 displayed warning stating: "Add RoadPoint nodes as children..."
- Selected RoadContainer1 and used "Create" menu to create a 2x2 road.
- Noted that RoadPoints were placed as children of RoadContainer1 object as expected. ("nodes" and "segments" nodes were no longer created.)
- Selected various RoadPoints, moved, and rotated. No errors. Redraw was consistent and reliable.
- Noted that RoadSegment bounding box could grow really large when start and end point rotations/positions were varied. This was not a defect. Just an unanticipated development.

TESTED EXPORT VARS:
- Selected RoadContainer1 and toggled "Create Geo" export variable off and on in the Inspector panel. Geometry disappeared and reappeared as expected.
- Toggled geo off. Then, moved and rotated RoadContainer1.
- Toggled geo on. Geometry was immediately drawn in the new location as expected.

TESTED ROADMANAGER:
- Continued testing using previously created custom scene.
- Added "RoadContainer2" as child of RoadContainer1. Then, dragged RoadPoints from RoadContainer1 into RoadContainer2. Many different error messages were displayed in output panel. Also, RoadContainer1 displayed its previous/default warning. Lastly, RoadContainer2 displayed a different warning saying: "A RoadContainer should have a RoadManager somewhere in its parent hierarchy or be the scene root." (So far, it seemed like things were working as expected.)
- Selected RoadContainer1 and added RoadManager object. No warnings were displayed on RoadManager.
- Dragged RoadContainer2 into RoadManager. RoadContainer2 warning went away. Although, geometry was not drawn.
- Selected RoadContainer2 and toggled "Create Geo" off and on in the Inspector panel. Geometry was redrawn as expected. No new errors were shown in the Output panel.

ADDITIONAL NOTES:
- Noted the inclusion of RoadIntersection base class.
- Noted that various nodes had been moved to the "nodes" folder in the project directory.
- Noted that various scripts had been moved to the "ui" folder.
- Noted addition of the "resources" folder containing .png images and .import files resulting in a cleaner general file structure.
- It might be a good idea to update the demo scene and commit the changes so newcomers don't encounter any unnecessary errors.
- Ran unit tests: Passing 12, Failing 0, Orphans 1.

CONCLUSION:
RoadManager, RoadContainer, and the rest of the supporting cast behaved reliably. Great work!

**Comment by TheDuckCow** [comment]
Thanks for reviewing @bdog2112!

> Noted that RoadSegment bounding box could grow really large when start and end point rotations/positions were varied. This was not a defect. Just an unanticipated development.

Yeah I also found this unexpected. I did quite a bit of searching and could not find a way to disable the editor bounding box. It's due to the fact that the mesh underneath in the node is actually counter-rotated differently to the one RoadPoint it's parented to, which indeed can lead to some large bounding boxes on selection. I find it somewhat distracting, but not terrible - and does at least make it crystal clear which meshes "belong" to a given RoadPoint.

> Added "RoadContainer2" as child of RoadContainer1. Then, dragged RoadPoints from RoadContainer1 into RoadContainer2. Many different error messages were displayed in output panel. Also, RoadContainer1 displayed its previous/default warning. Lastly, RoadContainer2 displayed a different warning saying: "A RoadContainer should have a RoadManager somewhere in its parent hierarchy or be the scene root." (So far, it seemed like things were working as expected.)

Some very good QA testing there, I didn't even think to put a roadcontainer as a child of another one! I suppose we should think how we want that to function. Maybe users would actually _want_ to create multiple levels of hierarchy actually. e.g. a road which has a bridge in the middle using a prefab scene, but the rest is using procedural geometry. It could be convenient that these "chunks" of the road could be grouped together, such that the child-most container rules apply the most (things like Road/Street name, properties like speed limit if we implement such things, etc). Might be hard to avoid the errors but something I'll take a look at, will merge this branch in the meantime.

**Pull request merged**